lib/puma/cluster.rb

```diff
@@ -28,10 +28,12 @@ def initialize(cli, events)
 
       @phased_state = :idle
       @phased_restart = false
+      @restart = false
     end
 
     def stop_workers
       log "- Gracefully shutting down workers..."
+      @launcher.close_binder_listeners unless @restart
```

Comment:
Some of these changes were related to leaked fd's, which is why I added the log output for it.  This may have been part of the fix of the macOS issue with ':reset' connections, which are now zero.  Previously they hovered around +/- 7.

Reply:
> I'm wondering what we were doing before that was wrong.

Remember your question about how some of the connections were being 'dumped' during a stop?  They were already accepted but then they were reset?  This fixes that, but we can't close them if a restart is being done, otherwise Binder#redirects_for_restart won't work because the sockets need to stay open.